magnum_cluster_api/resources.py

```diff
@@ -338,7 +338,7 @@ def get_object(self) -> dict:
                                 "allowVolumeExpansion": True,
                                 "kind": objects.StorageClass.kind,
                                 "metadata": {
-                                    "name": "share-%s"
+                                    "name": "share-nfs-%s"
```

Comment:
The good thing is that we can delete SC(StorageClass) in use without any issue.
So I'd prefer to check current SC's provisioner field and delete if that is old one(without proto prefix). And create a new one. And all those operations in the code, not manual.

Reply:
The upgrade is stuck with existing PVs that already have "provisioner" defined and which cannot be easily modified.

Reply:
It's in PVC on 
```
kind: PersistentVolumeClaim
metadata:
  annotations:
    ...
    volume.beta.kubernetes.io/storage-provisioner: manila.csi.openstack.org
    volume.kubernetes.io/storage-provisioner: manila.csi.openstack.org
    ...
```


And here in PV
```
apiVersion: v1
kind: PersistentVolume
metadata:
  annotations:
    pv.kubernetes.io/provisioned-by: manila.csi.openstack.org
spec:
  ...
  csi:
    ...
    driver: manila.csi.openstack.org
    ...
    volumeAttributes:
      ...
      storage.kubernetes.io/csiProvisionerIdentity: 1722446784123-8081-manila.csi.openstack.org
    ...
```